syntax = "proto3";

import "mixing_state.proto";
import "ls_state.proto";

package bayesmix;

message MarginalState {
  message ClusterState {
    oneof val {
      UnivLSState univ_ls_state = 1;
      MultiLSState multi_ls_state = 2;
      UnivDepLSSState univ_dep_ls_state = 3;
    }
    int32 cardinality = 4;
  }

  repeated ClusterState cluster_states = 1;
  repeated int32 cluster_allocs = 2 [packed = true];
  MixingState mixing_state = 3;
  int32 iteration_num = 4;
}
